server/events.go

```diff
@@ -336,6 +336,17 @@ func (ci *ClientInfo) forAssignmentSnap() *ClientInfo {
 	}
 }
 
+// forProposal returns the minimum amount of ClientInfo we need for assignment proposals.
+func (ci *ClientInfo) forProposal() *ClientInfo {
+	if ci == nil {
+		return nil
+	}
+	cci := *ci
```

Comment:
If we are considering these at scale, should we only copy what we need vs copying everything and removing which makes the allocation and GC system work a bit harder? Should we also consider a pool for these?